drivers/lora/sx12xx_common.c

```diff
@@ -50,24 +55,33 @@ int __sx12xx_configure_pin(const struct device * *dev, const char *controller,
 static void sx12xx_ev_rx_done(uint8_t *payload, uint16_t size, int16_t rssi,
 			      int8_t snr)
 {
-	Radio.Sleep();
-
 	dev_data.rx_buf = payload;
 	dev_data.rx_len = size;
 	dev_data.rssi = rssi;
 	dev_data.snr = snr;
 
+	if (MODEM_FREE(dev_data)) {
+		Radio.Sleep();
+	}
+
 	k_sem_give(&dev_data.data_sem);
 }
 
 static void sx12xx_ev_tx_done(void)
 {
-	Radio.Sleep();
+	if (MODEM_FREE(dev_data)) {
```

Comment:
We should probably consider making sx12xx_lora_send blocking and enter the sleep state from the send function. That would make the API easier to use since there is currently no way of knowing when the radio is ready short of polling it.

Reply:
That is the purpose of the follow on PR

Reply:
Ack. I saw it just after reviewing this one.